### ISSUE TITLE
VideoStreamDetail inconsistent casing with JSON RPC for hdrType and stereoMode keys

### ISSUE THREAD

**Issue by jurialmunkey** [xbmc/xbmc#22169]
### Describe the bug
Here is a clear and concise description of what the problem is:

In attempting to meet the requirements of the [new InfoTagVideo setter methods](https://github.com/xbmc/xbmc/pull/19459) I have discovered an inconsistency between naming in JSON RPC and the VideoStreamDetail class.

The python VideoStreamDetail() class requires camelCase kwargs for hdrType and stereoMode. However, JSON RPC outputs stream detail keys all lowercase as hdrtype and stereomode

This results in a TypeError being raised when attempting to pass library database stream details from JSON RPC back to the relevant class for the newly required InfoTagVideo setter methods. 

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

It should be possible to pass the stream details dictionary retrieved via JSON RPC by directly expanding the dictionary as VideoStreamDetail(**dictionary) rather than converting the key names via explicitly naming.

## Actual Behavior

The casing of the python VideoStreamDetail class kwargs and the JSON RPC datapoint do not match so a TypeError is raised

## Possible Fix

Either VideoStreamDetail() method accepts the kwargs using the lowercase name returned by JSON RPC or JSON RPC returns the keys with the same camelCase as required by the method.

### To Reproduce

Get [Video.Streams](https://kodi.wiki/view/JSON-RPC_API/v12#Video.Streams) object via "VideoLibrary.GetMovieDetails" using xbmc.executeJSONRPC() and then attempt to pass back to the stream object class by expanding the dictionary

``` python
import xbmc
from json import dumps, loads

movieid = 1
query = {
    "jsonrpc": "2.0",
    "method": "VideoLibrary.GetMovieDetails",
    "params": {"movieid": movieid, "properties": ["streamdetails"]},
    "id": 1}
jrpc_response = loads(xbmc.executeJSONRPC(dumps(query)))
streamdetails = jrpc_response.get('result').get('moviedetails').get('streamdetails')


for i in streamdetails.get('video'):
    sd = VideoStreamDetail(**i)   # To create object for InfoTagVideo.addVideoStream()
```

### Debuglog

The debuglog can be found here:

[kodi.log](https://github.com/xbmc/xbmc/files/10048295/kodi.log)

Log is not particularly relevant other than for an example output from JSONRPC for streamdetails

```
{'audio': [{'channels': 2, 'codec': 'aac', 'language': ''}, {'channels': 1, 'codec': 'aac', 'language': 'eng'}], 'subtitle': [{'language': 'eng'}, {'language': 'hun'}, {'language': 'ger'}, {'language': 'fre'}, {'language': 'spa'}, {'language': 'ita'}, {'language': 'jpn'}, {'language': ''}], 'video': [{'aspect': 1.7777780294418335, 'codec': 'h264', 'duration': 46, 'hdrtype': '', 'height': 576, 'language': '', 'stereomode': '', 'width': 1024}]}
```

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

Whilst it is possible to workaround this issue by explictly naming each key in the setter method, it is still a workaround. The key names between JSON RPC and the corresponding setter method should match to ensure future proofing if additional keys are added without requiring all addons to update.

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] OSX
 - [x] Windows
 - [ ] Windows UWP

 - Operating system version/name: Windows 11
 - Kodi version: Nexus Nightly

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by joseluismarti** [comment]
Right, a TypeError occurs when trying to pass the library movie details from JSON RPC to the VideoStreamDetail method.

This can be solved by changing these two keys to lower case in the VideoStreamDetail method to match those used in the JSON response. But it would mean modifying the API.

Another possible solution would be to add a second constructor with lowercase keys.

**Comment by fuzzard** [comment]
@Montellese whats the best way to handle this in your opinion, change json rpc api, or change the python api?

**Comment by Montellese** [comment]
The main problem is that the whole Python API uses camelCase while the whole JSON-RPC API uses camelcase. So if we change one of the APIs for this specific instance we will cause an inconsistency with the rest of the API independent of the fact that we would also introduce a backwards incompatible change.

Furthermore from a security point of view I would always prefer some kind of input validation instead of just passing on data received from one API to another API.

**Comment by CastagnaIT** [comment]
I agree, since are two different interfaces its normal that can have different operations or result,
then also it is not mandatory to have the same naming conventions

if it becomes necessary to make a change
I would prefer to start introduce camelcase on JSON-RPC interface
that IMO its more readable instead to have all lowercase
and may have even less impact on breaking change
because modifying the Python side is sure to break many addons (and scripts like scrapers)

**Comment by jurialmunkey** [comment]
In terms of mixedCase in the python API for variable names - that's never really been Pep8 compliant:
https://peps.python.org/pep-0008/#function-and-variable-names

mixedCase was only ever for legacy reasons as the "prevailing style" but the trend everywhere has been to deprecate if possible. Even the example of "prevailing style" in the pep8 docs (threading.py) has deprecated mixedCase whilst using function wrappers to maintain backwards compatibility e.g.
https://github.com/python/cpython/blob/6d8da238ccdf946dc90e20821652d8caa25b76ba/Lib/threading.py#L1486-L1505

The main reason I raised the issue is because these are new parts of the API and the methods being replaced *did* accept lowercase. All my naming of infolabels is lowercase because that's how infolabels have always been defined in the docs.

It's easy enough to work around either way, just seems strange to me to enforce a non-pep8-compliant naming convention in the new method when the previous method it is replacing accepted lowercase naming conventions (which was both pep8 compliant and matched with the jsonrpc schema).

**Comment by CastagnaIT** [comment]
someone else had mentioned to use pep8 in the kodi python interface/modules in the past,
but if i remember correctly i had read that it is a choice due to because we have a C++ conversion problem

**Comment by jurialmunkey** [comment]
I don't want to get too derailed discussing pep8 or snake_case as that's not my concern. My point is that moving infolabels to be stored as mixedCase goes against the majority of the existing naming conventions.

Prior to Nexus, the python API docs specified lowercase names for infolabels in dictionaries as seen in the setInfo() method I linked above. Storing infolabels in mixedCase is a new requirement.

SQL databases are not case-sensitive by default and the convention is for lowercase names. My expectation is that interfaces to/from the database follow that lowercase naming convention (as we see in the JSON RPC methods and the *old* python setters).

**Comment by Montellese** [comment]
OK looks like that's an oversight on my part. Usually all parameters in C++ code are camelCase. What is the general opinion on this? Since the new setters have been introduced in Nexus Alpha 1 they haven't been part of a stable release yet. Can we still change them at this point in the release cycle to match the previous behaviour and the JSON-RPC API?

@jurialmunkey do you happen to have a list of parameters which would need to be adjusted? To me it looks like only the two mentioned parameters `hdrType` and `stereoMode` are affected.

**Comment by fuzzard** [comment]
@Montellese my opinion is we haven't done a formal release yet, so let's fix it before final Nexus release

**Comment by jurialmunkey** [comment]
@Montellese - For VideoStreamDetails it is just those two (and none for Audio/Subs) but there's quite a few mixedCase in InfoTagVideo where the equivalent for setInfo is lowercase.

Less likely to come up as an issue in InfoTagVideo since most are individual methods where typically a single unnamed arg will be passed rather than a named keyword - but technically the names differ

e.g. [InfoTagVideo.setLastPlayed](https://xbmc.github.io/docs.kodi.tv/master/kodi-dev-kit/group__python___info_tag_video.html#ga1cc438431e9ac79ae0217433daf61826) asks for `lastPlayed` but [ListItem.setInfo](https://xbmc.github.io/docs.kodi.tv/master/kodi-dev-kit/group__python__xbmcgui__listitem.html#ga0f1e91e1d5aa61d8dd0eac90e8edbf18) asks for `lastplayed`


These are all the InfoTagVideo names I spotted where the new method has a mixedCase named keyword but the setInfo has a lowercase one:

dbId
sortSeason
episodeGuide
setId
trackNumber
plotOutline
originalTitle
sortTitle
tagLine
tvshowTitle
tvshowStatus (this one is usually just `status`)
setOverview
productionCode (this one is usually just `code`)
firstAired (bit confused by this one - I think it matches `premiered` but possibly could be `aired` )
lastPlayed
filenameAndPath
imdbNumber
dateAdded
mediaType
showLinks

**Comment by Montellese** [comment]
@jurialmunkey I can change those but it will only affect the documentation and will have zero effect on the actual code which is why I didn't list them. I'll try to get a PR done sometime this week but no promises since it's an extremely busy time at work :-/

**Comment by Montellese** [comment]
See #22192.
